app/questionnaire/questionnaire_manager.py

```diff
@@ -125,3 +126,44 @@ def get_schema_item_by_id(self, item_id):
 
     def get_schema(self):
         return self._schema
+
+    def add_answer(self, block, question, post_data, answer_store):
+        if self.state is None:
+            self.process_incoming_answers(block, post_data)
+
+        question_schema = self._schema.get_item_by_id(question)
+        question_state = self.state.find_state_item(question_schema)
+
+        answer_schema = question_schema.answers[0]  # Single answer for now.
+        new_answer = self._create_new_answer(answer_schema, question_state, answer_store)
+
+        question_answers = question_state.children
+        question_answers.append(new_answer)
+
+        self.update_questionnaire_store(block)
+
+    @staticmethod
+    def _create_new_answer(answer_schema, question_state, answer_store):
+        new_answer = answer_schema.construct_state()
+
+        existing = answer_store.filter(answer_id=answer_schema.id)
+        last_answer = existing[-1:]
+        next_instance_id = 0 if len(last_answer) == 0 else int(last_answer[0]['answer_instance']) + 1
+        new_answer_schema = copy.deepcopy(new_answer.schema_item)
+        new_answer_schema.widget.name += '_' + str(next_instance_id)
+        new_answer.schema_item = new_answer_schema
+        new_answer.parent = question_state
+        new_answer.instance = next_instance_id
+        return new_answer
+
+    def remove_answer(self, block, post_data, answer_store):
+        if self.state is None:
+            self.process_incoming_answers(block, post_data)
+
+        index_to_remove = post_data.get('action[remove_answer]')
```

Comment:
maybe we can pass the index through to this function? IMO references to the name of a button should probably by up at the 1st level rather than being extracted lower down.